doc/src/index-fileformats.md

```diff
@@ -0,0 +1,90 @@
+# Tantivy index file formats
+This document defines the index file formats used in Tantivy. 
+
+Tantivy has one metadata json file for each index and a few files for each segment, each file storing a specific datastructure.
+
+## Index metadata json
+For a given index, Tantivy stores the following metadata in a json file `meta.json` :
```

Comment:
Linking to the code can also help to familiarize with the code base
https://github.com/tantivy-search/tantivy/blob/fef428a9c692a2c1cf3242642657695c71dbc0c1/src/core/index_meta.rs#L206-L216

Generally I think it would be nice to have this information in the rust docs and referenced here, so it's unlikely to be outdated

Reply:
yes, for the metadata file it's really relevant as the code almost auto document it.

Reply:
@PSeitz I would actually use a github link to avoid pointing to a specific version such as this link: https://docs.rs/tantivy/0.14.0/tantivy/